client/state/activity-log/reducer.js

```diff
@@ -25,9 +26,23 @@ export const filterState = ( state = emptyFilter, { type, filter } ) => {
 	}
 };
 
+export const requestFreshness = ( state = defaultRequestFreshness, { type } ) => {
+	switch ( type ) {
+		case ACTIVITY_LOG_FILTER_SET:
+			return 100;
+
+		case ACTIVITY_LOG_FILTER_UPDATE:
+			return defaultRequestFreshness;
+
+		default:
+			return state;
+	}
+};
```

Comment:
this seems curious…what are we trying to accomplish with this?

Reply:
i'm trying to figure out how to reset the freshness of the API request to fetch acitivity

Reply:
this would be a good case for a wrapper component or triggering updates in `componentDidUpdate()`

```js
componentDidUpdate( prevProps ) {
	if ( ! isEqual( prevProps.filter, this.props.filter ) ) {
		this.props.refreshLogs();
	}
}
```

or separately we could store the key as the serialization of the arguments

```js
() => {
	const id = `activity-log-${ siteId }-${ JSON.stringify( filter ) }`;
	return requestHttpData(
		id,
		http( … ),
		…
	);
}
```

there are a number of ways of doing this

Reply:
thanks! i'll try out the serialized key idea. :)  - I love all your suggestions. Thanks for taking the time for reviewing :)

Reply:
I ended up using the serialized filter in the request id - it was more in-line with how were are fetching.